alkira/resource_alkira_peering_gateway_cxp.go

```diff
@@ -0,0 +1,173 @@
+package alkira
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/alkiranet/alkira-client-go/alkira"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceAlkiraPeeringGatewayCxp() *schema.Resource {
+	return &schema.Resource{
+		Description:   "Manage CXP Peering Gateways.",
+		CreateContext: resourceAlkiraPeeringGatewayCxpCreate,
+		ReadContext:   resourceAlkiraPeeringGatewayCxpRead,
+		UpdateContext: resourceAlkiraPeeringGatewayCxpUpdate,
+		DeleteContext: resourceAlkiraPeeringGatewayCxpDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Description: "The name of the Peering Gateway.\nOnce deployed this property can not be changed. All the changes will be ignored.",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"description": {
+				Description: "Description of the Peering Gateway.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"cxp": {
+				Description: "The CXP to which the Gateway is attached.\nOnce deployed this property can not be changed. All the changes will be ignored.",
```

Comment:
same here... we simple can't say that "all changes will be ignored"... API should return proper error...

Reply:
We had a discussion during the ACG meeting, basically if a resource cannot be updated it would just ignore the changes, if the update was conditional , it would throw an error. 
So in this case the changes would be just ignored. I was asked to add this note in the documentation.

Reply:
I disagree. This doesn't belong to ACG team. It's just a bad user experience. If backend doesn't want to fix this, please do validation in `update` function.

Reply:
Even we wanted to have this description. It should be like this and have line break at 80 like this:

```
Description: "The CXP to which the Gateway is attached to. " +
                      "Once deployed, this field can't be updated.",
```